packages/@webex/plugin-meetings/src/meeting/index.ts

```diff
@@ -3348,193 +3351,177 @@ export default class Meeting extends StatelessWebexPlugin {
-    this.mediaProperties.setLocalVideoTrack(localTrack);
+    this.mediaProperties.setLocalVideoStream(localStream);
 
-    this.video.handleLocalTrackChange(this);
+    this.video.handleLocalStreamChange(this);
 
-    localTrack?.on(LocalTrackEvents.Muted, this.localVideoTrackMuteStateHandler);
-    localTrack?.on(LocalTrackEvents.UnderlyingTrackChange, this.underlyingLocalTrackChangeHandler);
+    localStream?.on(StreamEventNames.MuteStateChange, this.localVideoStreamMuteStateHandler);
+    localStream?.on(LocalStreamEventNames.OutputTrackChange, this.localOutputTrackChangeHandler);
 
-    if (!this.isMultistream || !localTrack) {
-      // for multistream WCME automatically un-publishes the old track when we publish a new one
-      await this.unpublishTrack(oldTrack);
+    if (!localStream) {
+      await this.unpublishStream(MediaType.VideoMain);
     }
-    await this.publishTrack(this.mediaProperties.videoTrack);
+    await this.publishStream(this.mediaProperties.videoStream, MediaType.VideoMain);
   }
 
   /**
-   * Stores the reference to a new screen share video track, sets up the required event listeners
-   * on it, cleans up previous track, etc.
+   * Stores the reference to a new screen share stream, sets up the required event listeners
+   * on it, cleans up previous stream, etc.
+   * It also sends the floor grant/release request.
    *
-   * @param {LocalDisplayTrack | undefined} localDisplayTrack local camera track
+   * @param {LocalDisplayStream | undefined} localDisplayStream local display stream
    * @returns {Promise<void>}
    */
-  private async setLocalShareVideoTrack(localDisplayTrack?: LocalDisplayTrack) {
-    const oldTrack = this.mediaProperties.shareVideoTrack;
+  private async setLocalShareVideoStream(localDisplayStream?: LocalDisplayStream) {
+    const oldStream = this.mediaProperties.shareVideoStream;
 
-    oldTrack?.off(LocalTrackEvents.Ended, this.handleShareVideoTrackEnded);
-    oldTrack?.off(LocalTrackEvents.UnderlyingTrackChange, this.underlyingLocalTrackChangeHandler);
+    oldStream?.off(StreamEventNames.Ended, this.handleShareVideoStreamEnded);
+    oldStream?.off(LocalStreamEventNames.OutputTrackChange, this.localOutputTrackChangeHandler);
 
-    this.mediaProperties.setLocalShareVideoTrack(localDisplayTrack);
+    this.mediaProperties.setLocalShareVideoStream(localDisplayStream);
 
-    localDisplayTrack?.on(LocalTrackEvents.Ended, this.handleShareVideoTrackEnded);
-    localDisplayTrack?.on(
-      LocalTrackEvents.UnderlyingTrackChange,
-      this.underlyingLocalTrackChangeHandler
+    localDisplayStream?.on(StreamEventNames.Ended, this.handleShareVideoStreamEnded);
+    localDisplayStream?.on(
+      LocalStreamEventNames.OutputTrackChange,
+      this.localOutputTrackChangeHandler
     );
 
-    this.mediaProperties.mediaDirection.sendShare = this.mediaProperties.hasLocalShareTrack();
+    this.mediaProperties.mediaDirection.sendShare = !!localDisplayStream;
```

Comment:
why are we not using hasLocalShareStream() here?

Reply:
Sorry. Must''ve come from [this commit](https://github.com/brycetham/webex-js-sdk/blob/26d52f3bdc66538b20bc7a2956deb291dbd1fadb/packages/%40webex/plugin-meetings/src/meeting/index.ts#L3445) while solving merge conflicts.

Updated this in #3053